subsys/net/lib/zperf/zperf_tcp_uploader.c

```diff
@@ -124,6 +124,13 @@ int zperf_tcp_upload(const struct zperf_upload_params *param,
 		return sock;
 	}
 
+	if (zsock_setsockopt(sock, IPPROTO_TCP, TCP_NODELAY,
```

Comment:
Let's call this only if `tcp_nodelay` is set. An offloaded socket implementation may not support it, and that way it'll only fail if the `-n` parameter is provided, not always.

Reply:
or I can put this under `CONFIG_NET_NATIVE_TCP` ?

Reply:
No, one implementation may support it other not. I really se no point setting a socket option if it has no effect, other than potential failure.

Reply:
ok, added both, as the internal TCP_NODELAY is also protected by CONFIG_NET_NATIVE_TCP

Reply:
>  as the internal TCP_NODELAY is also protected by CONFIG_NET_NATIVE_TCP

Sorry, I don't follow, what's protected where?

I still maintain you should not use CONFIG_NET_NATIVE_TCP here, it only brings confusion. Now `-n` parameter cannot be used at all by any offloaded socket implementation, silently ignoring the option. You should only check if the `-n` parameter was provided before setting the option
* if the socket implementation supports it, then it just works
* otherwise, user gets a clear error message, that he should not use the option

Reply:
I was referring to the Zephyr neworking stack code for TCP_NODELAY being protected by CONFIG_NET_NATIVE_TCP, but I understand your point that `set_sockopt` might itself be offloaded and they might support , will change and push.